cmd/blockstorage_snapshot_list.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/exoscale/cli/pkg/globalstate"
+	"github.com/exoscale/cli/pkg/output"
+	v3 "github.com/exoscale/egoscale/v3"
+)
+
+type blockstorageSnapshotListItemOutput struct {
+	ID     v3.UUID     `json:"id"`
+	Name   string      `json:"name"`
+	Zone   v3.ZoneName `json:"zone"`
+	Volume v3.UUID     `json:"volume"`
+}
+
+type blockstorageSnapshotListOutput []blockstorageSnapshotListItemOutput
+
+func (o *blockstorageSnapshotListOutput) ToJSON()  { output.JSON(o) }
+func (o *blockstorageSnapshotListOutput) ToText()  { output.Text(o) }
+func (o *blockstorageSnapshotListOutput) ToTable() { output.Table(o) }
+
+type blockstorageSnapshotListCmd struct {
+	cliCommandSettings `cli-cmd:"-"`
+
+	_ bool `cli-cmd:"list"`
+
+	Zone v3.ZoneName `cli-short:"z" cli-usage:"zone to filter results to"`
+}
+
+func (c *blockstorageSnapshotListCmd) cmdAliases() []string { return gCreateAlias }
+
+func (c *blockstorageSnapshotListCmd) cmdShort() string { return "List Block Storage Volumes" }
+
+func (c *blockstorageSnapshotListCmd) cmdLong() string {
+	return fmt.Sprintf(`This command lists Block Storage Volumes.
+
+Supported output template annotations: %s`,
+		strings.Join(output.TemplateAnnotations(&blockstorageListOutput{}), ", "))
+}
+
+func (c *blockstorageSnapshotListCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
+	return cliCommandDefaultPreRun(c, cmd, args)
+}
+
+func (c *blockstorageSnapshotListCmd) cmdRun(_ *cobra.Command, _ []string) error {
+	client := globalstate.EgoscaleV3Client
+	ctx := gContext
+
+	resp, err := client.ListZones(ctx)
+	if err != nil {
+		return err
+	}
+	zones := resp.Zones
+
+	if c.Zone != "" {
+		endpoint, err := client.GetZoneAPIEndpoint(ctx, c.Zone)
+		if err != nil {
+			return err
+		}
+
+		zones = []v3.Zone{{APIEndpoint: endpoint}}
+	}
+
+	output := make(blockstorageSnapshotListOutput, 0)
+	for _, zone := range zones {
+		c := client.WithEndpoint(zone.APIEndpoint)
+
+		resp, err := c.ListBlockStorageSnapshots(ctx)
+		if err != nil {
+			// TODO: remove it once Block Storage is deployed in every zone.
+			if strings.Contains(err.Error(), "Availability of the block storage volumes") {
+				continue
+			}
+
+			return err
```

Comment:
I think we will forget to do this and it's a headache. How about we just log the error and `continue`. Besides I don't think it's good usability to return here if there is an error. What if the API is unstable in one zone, let's continue and list the other zones for the user.

same recommendation for the volume command.(you added the same TODO there)

Reply:
I agree this not super convenient.. And of course temporary! But for the moment if you do an `exo compute block-storage list` without giving a zone, by default we try to list in all zone like every other command, how do you ignore not working/deployed zone in this case?